### DISCUSSION TITLE
How do I get PixelFormat?

### DISCUSSION THREAD

**Discussion by Toxa58rus** [SixLabors/ImageSharp#1690] [Q&A]
I get a stream of bytes, I know it's an image. I need to process the pixels of this image and save it in its original format. How can I get the PixelFormat before loading the image using Load? To use the pixel manipulation feature, I need to use Load <TPixel>, but I don't know which TPixel to use.

**Comment by JimBobSquarePants** [comment]
You’re looking for [Identify](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_Identify_SixLabors_ImageSharp_Configuration_System_Byte___SixLabors_ImageSharp_Formats_IImageFormat__) however the generic pixel format parameter is simply an indication of how you want the decoded pixel data to be arranged in memory, not how the pixel format of the encoded image. This means it can be whatever you want it to be.

**Reply by Toxa58rus** [reply]
That is, it only affects how the pixels are aligned, and I choose the alignment format that suits me. Did I understand correctly?

**Reply by JimBobSquarePants** [reply]
Yep, exactly
